File: Bubble_Sort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:  # Swap if the element is greater
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:  # Stop if already sorted
            break

# Taking input from the user
n = int(input("Enter the number of elements: "))
arr = list(map(int, input("Enter the elements separated by spaces: ").split()))

print("Unsorted array:", arr)
bubble_sort(arr)
print("Sorted array:", arr)

File: Bubble_Sort.c
//Comparison Sorting : Bubble Sorting (C-Implementation)

#include<stdio.h>
#include<windows.h>

void bubbleSort(int arr[], int n)
{
    int i, j, temp;
    for (i = 0; i < n - 1; i++) 
	{
        for (j = 0; j < n - i - 1; j++) 
		{
            if (arr[j] > arr[j+1]) 
			{
                //Swap arr[j] and arr[j+1] if value of arr[j] is more than the value of arr[j+1] to make the highest number at the end
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

int main() 
{
    int n, i;

    printf("Enter the number of elements: ");
    scanf("%d", &n);

//    int arr[n]; //OR 
    int *arr = (int*)malloc(n*sizeof(int)); //to dynamically allocate the memory
    if(arr==NULL) 
	{
        printf("Memory allocation failed!");
        return 1; 
    }

    printf("Enter the elements:\n");
    for(i = 0; i < n; i++) 
	{
        scanf("%d", &arr[i]);
    }

    bubbleSort(arr, n); //Function calling to sort the elements
	
	//To display the sorted elements
    printf("Sorted array elements are :\n"); 
    for(i = 0; i < n; i++)
    {
    	printf("%d ",arr[i]);
	}
	free(arr); //to free the allocated memory

    return 0; 
}